GPU/Common/FramebufferCommon.cpp

```diff
@@ -40,16 +40,44 @@ void CenterRect(float *x, float *y, float *w, float *h, float origW, float origH
 		outW = frameW;
 		outH = frameH;
 	} else {
-		// Add special case for 1080p displays, cutting off the bottom and top 1-pixel rows from the original 480x272.
-		// This will be what 99.9% of users want.
-		if (origW == 480 && origH == 272 && frameW == 1920 && frameH == 1080 && !rotated) {
-			*x = 0;
-			*y = -4;
-			*w = 1920;
-			*h = 1088;
-			return;
+		bool fullScreenZoom = true;
+#ifdef _WIN32
+		fullScreenZoom = g_Config.bFullScreen;
+#endif
+		if (fullScreenZoom && g_Config.iSmallDisplayZoom != 0) {
+			if (g_Config.iSmallDisplayZoom != 1) {
+				float offsetX = (g_Config.fSmallDisplayOffsetX - 0.5f) * 2.0f * frameW;
+				float offsetY = (g_Config.fSmallDisplayOffsetY - 0.5f) * 2.0f * frameH;
+				float customZoom = g_Config.fSmallDisplayCustomZoom / 8.0f;
```

Comment:
Because I use it also for image scale which is smallest possible while keeping psp proportions:)